File: solutions/java/leap/1/src/main/java/Leap.java
/**
  * The Leap class defines a method that tests whether a given year is a leap year.
  * 
  * Rules to qualify:
  *   year is evenly divisble by 4 but not by 100 or year is evenly divisible by 400.
  */

class Leap {

    /**
      * Tests year divisibility and returns whether conditions for leap year are met.
      *
      * @param year the year being tested
      * @return true if given year qualifies as a leap year,
      *         false if given year dos not qualify as a leap year
      */
    boolean isLeapYear(int year) {
        return (year % 4 == 0 && year % 100 != 0) || year % 400 == 0;
    }

}

File: solutions/python/leap/3/leap.py
def leap_year(year):
    """Tests whether a given year qualifies as a leap year.
    
    :param year: the year being tested
    :return: True if the year qualifies as a leap year,
    False if the year does not qualify as a leap year"""
    if year % 4 == 0 and year % 100 != 0:
        return True
    if year % 400 == 0:
        return True
    return False